### ISSUE TITLE
Fullscreen Detection broken

### ISSUE THREAD

**Issue by netAction** [4ian/GDevelop#6477]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/4ian/GDevelop/issues)

### Describe the bug

Affects: HTML5 export.

Both fullscreen detection and fullscreen switch methods are broken when the user leaves fullscreen on Chrome/Windows via Esc key or on Chrome/Android via < button or on Safari/IOs via swipe down.

After executing the _Activate fullscreen_ action, GDevelop thinks it is in fullscreen. Even if it is not, because fullscreen has been left manually. Then running _Activate fullscreen_ again might crash. And worse, the condition _The game is in fullscreen_ says true but it is not.

The Javascript code _if (document.fullscreenElement != null)_ returns the correct value on all three browsers, no matter if the fullscreen has been closed via _Activate fullscreen_ action or via the intended browser way.

Minimal project for reproduction:
<img width="1062" alt="Screenshot 2024-03-21 223432" src="https://github.com/4ian/GDevelop/assets/1829847/0ed91c91-c1b6-42dc-b673-4ad9295554fe">

### Steps to reproduce

1. Create a button and bind _Activate fullscreen_ on click.
2. Leave Fullscreen in the way your browser suggests.
3. Check  _The game is in fullscreen_ or use the _Activate fullscreen_ action again. Both don't work correctly.
4. Check _if (document.fullscreenElement != null)_. It does work.

### GDevelop platform

Desktop

### GDevelop version

5.3.195

### Platform info

IDE: GDevelop via Microsoft Store on Windows 11

Browser: Chrome/Windows, Chrome/Android, Safari/IOs

**Comment by ClementPasteau** [comment]
Hi @netAction does this happen when you publish on gd.games too? 
If so, can you share a link so we can see the problem in action?
Thanks

**Comment by netAction** [comment]
Yes the bug is reproducable in gd.games. https://gd.games/netaction/fullscreen-bug

![Screenshot 2024-03-28 100308](https://github.com/4ian/GDevelop/assets/1829847/63324999-6d0b-404f-ae03-1119e8b55247)

![Screenshot 2024-03-28 100334](https://github.com/4ian/GDevelop/assets/1829847/f789d7cd-f792-4ba1-8677-30ae705d5616)

**Comment by ClementPasteau** [comment]
Thanks, that's very helpful and detailed, we're probably missing an event not being listened properly or using a wrong logic

**Comment by necaTecnico** [comment]
Hi, I am a university student and would like to solve the https://github.com/4ian/GDevelop/issues/2162  bug for a university project. This bug mentioned seems to be the same as the one here (https://github.com/4ian/GDevelop/issues/6477). I have been looking at the code and trying to do all the suggestions mentioned in the other issue as well as this one. but nothing really worked. Anyone got any recommendations or help please? I didn't really understand the proposed solution as well as this workaround, as I didn't get where in the code to place these snippets.

**Comment by necaTecnico** [comment]
@netAction @netAction Hi, I am a university student and would like to solve the https://github.com/4ian/GDevelop/issues/2162 bug for a university project. This bug mentioned seems to be the same as the one here (https://github.com/4ian/GDevelop/issues/6477). I have been looking at the code and trying to do all the suggestions mentioned in the other issue as well as this one. but nothing really worked. Anyone got any recommendations or help please? I didn't really understand the proposed solution as well as this workaround, as I didn't get where in the code to place these snippets.

**Comment by necaTecnico** [comment]
@ClementPasteau  @netAction Hi, I am a university student and would like to solve the https://github.com/4ian/GDevelop/issues/2162 bug for a university project. This bug mentioned seems to be the same as the one here (https://github.com/4ian/GDevelop/issues/6477). I have been looking at the code and trying to do all the suggestions mentioned in the other issue as well as this one. but nothing really worked. Anyone got any recommendations or help please? I didn't really understand the proposed solution as well as this workaround, as I didn't get where in the code to place these snippets.

**Comment by necaTecnico** [comment]
@ClementPasteau Ok, but let me just check with my professor if we can open PRs before being reviewed by him first. Thanks a lot for the quick reach out. I will get back to you.

But just one question, this issue looks the same to me as this one (https://github.com/4ian/GDevelop/issues/2162) right? Because I actually chose this linked bug here to solve first, but then I found this one too which looks to be the same problem.

**Comment by necaTecnico** [comment]
Hi @ClementPasteau, here is what I have changed in this file so far, following your recommendations and other in this issue and in https://github.com/4ian/GDevelop/issues/2162, however it is still not working.... also I cannot see logs when I do them. Any recommendations on what I have done, and also, is there any specific way required to compile the code after changing it that I need to do before npm start? 

PS: Sorry I can't open a PR before having the solution for the bug because of the project's evaluation my teacher has set. So here is the file converted to txt.

[runtimegame-pixi-renderer.txt](https://github.com/user-attachments/files/19216849/runtimegame-pixi-renderer.txt)

**Comment by ClementPasteau** [comment]
Hi @necaTecnico, I want to help, though it's really not efficient like this.
Github branches and PRs are made for this kind of back and forth discussions, so we can give you feedback and iterate. This is how it works in the open source world :)
Unless you have a PR opened, I will not be able to help you. (Try convincing your teacher!)

**Comment by necaTecnico** [comment]
ok @ClementPasteau I will try, but just tell me one thing, so while I wait for a response back I can do something..... is there any specific way required to compile the code after changing it that I need to do before npm start (so I can make sure I am testing my changes)? Also do I need to do something to activate logs on chrome console? because none show up.

**Comment by necaTecnico** [comment]
Hey @ClementPasteau, I have opened a pull request related to this issue. It's currently awaiting review and approval. Please take a look when possible, so we can discuss further. Thank you!

**Comment by necaTecnico** [comment]
Hi @ClementPasteau, just looping back here following our earlier exchange where you suggested opening a PR for proper discussion.

I've since opened PR #7538  with a proposed fix and some testing groundwork. While the solution doesn't fully resolve the issue across all browsers, I wanted to share it as a concrete basis for discussion and iteration.

Would appreciate your thoughts whenever you get a chance — especially to understand if I'm on the right track or if you'd approach this differently.

Thanks again!
@necaTecnico
